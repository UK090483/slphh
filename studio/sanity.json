{
  "root": true,
  "project": {
    "name": "perspectivRegion"
  },
  "api": {
    "projectId": "d85o021b",
    "dataset": "production"
  },
  "plugins": [
    "@sanity/base",
    "@sanity/components",
    "@sanity/default-layout",
    "@sanity/default-login",
    "@sanity/desk-tool",
    "@sanity/production-preview",
    "media",
    "iframe-pane"
  ],
  "env": {
    "development": {
      "plugins": ["@sanity/vision"]
    }
  },
  "parts": [
    {
      "name": "part:@sanity/base/schema",
      "path": "./schemas/schema"
    },
    {
      "name": "part:@sanity/desk-tool/structure",
      "path": "./desk/deskStructure.js"
    },
    {
      "implements": "part:@sanity/production-preview/resolve-production-url",
      "path": "./parts/resolveProductionUrl.js"
    },
    {
      "name": "part:@sanity/base/initial-value-templates",
      "path": "./parts/initialValueTemplates.js"
    }
  ]
}
