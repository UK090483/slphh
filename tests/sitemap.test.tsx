import { ServerResponse } from "http";
import { getServerSideProps } from "../pages/sitemap.xml";
import * as Sanity from "@lib/SanityService/sanity.server";

jest.mock("@lib/SanityService/sanity.server");

describe("Sitemap", () => {
  it("should render no Lang ", async () => {
    const getClientMock = jest.spyOn(Sanity, "getSanityClient");
    getClientMock.mockReturnValue({
      //@ts-ignore
      fetch: async () => [{ slug: "testSlug1" }, { slug: "testSlug2" }],
    });

    jest.useFakeTimers("modern");
    jest.setSystemTime(new Date(2020, 3, 1));
    let smRes: any;
    const res = {
      setHeader: () => {},
      write: (sm: string) => (smRes = sm),
      end: () => {},
    } as unknown as ServerResponse;
    //@ts-ignore
    await getServerSideProps({ res });
    expect(smRes).toMatchSnapshot();
  });
});
