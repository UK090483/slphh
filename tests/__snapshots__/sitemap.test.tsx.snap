// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sitemap should render no Lang  1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:news=\\"http://www.google.com/schemas/sitemap-news/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\" xmlns:image=\\"http://www.google.com/schemas/sitemap-image/1.1\\" xmlns:video=\\"http://www.google.com/schemas/sitemap-video/1.1\\"><url><loc>https://www.perspektivregion.eu/testSlug1</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"de\\" href=\\"https://www.perspektivregion.eu/testSlug1\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"da\\" href=\\"https://www.perspektivregion.eu/daundefined\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"en\\" href=\\"https://www.perspektivregion.eu/enundefined\\"/></url><url><loc>https://www.perspektivregion.eu/daundefined</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"de\\" href=\\"https://www.perspektivregion.eu/testSlug1\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"da\\" href=\\"https://www.perspektivregion.eu/daundefined\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"en\\" href=\\"https://www.perspektivregion.eu/enundefined\\"/></url><url><loc>https://www.perspektivregion.eu/enundefined</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"de\\" href=\\"https://www.perspektivregion.eu/testSlug1\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"da\\" href=\\"https://www.perspektivregion.eu/daundefined\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"en\\" href=\\"https://www.perspektivregion.eu/enundefined\\"/></url><url><loc>https://www.perspektivregion.eu/testSlug2</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"de\\" href=\\"https://www.perspektivregion.eu/testSlug2\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"da\\" href=\\"https://www.perspektivregion.eu/daundefined\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"en\\" href=\\"https://www.perspektivregion.eu/enundefined\\"/></url><url><loc>https://www.perspektivregion.eu/daundefined</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"de\\" href=\\"https://www.perspektivregion.eu/testSlug2\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"da\\" href=\\"https://www.perspektivregion.eu/daundefined\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"en\\" href=\\"https://www.perspektivregion.eu/enundefined\\"/></url><url><loc>https://www.perspektivregion.eu/enundefined</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"de\\" href=\\"https://www.perspektivregion.eu/testSlug2\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"da\\" href=\\"https://www.perspektivregion.eu/daundefined\\"/><xhtml:link rel=\\"alternate\\" hreflang=\\"en\\" href=\\"https://www.perspektivregion.eu/enundefined\\"/></url></urlset>"`;
